artemis-service-extensions/src/main/java/org/apache/activemq/artemis/service/extensions/xa/recovery/XARecoveryConfig.java

```diff
@@ -45,16 +45,41 @@
    private final Map<String, String> properties;
    private final ClientProtocolManagerFactory clientProtocolManager;
 
+   // ServerLocator properties
```

Comment:
why you had to bring those here?

Reply:
@clebertsuconic I was trying to make them available in createServerLocator() so that I can apply them to new instances of ServerLocator. Would it be better to hold a reference to original locator/factory instead? Or is there a better way to pass those properties I missed?

Reply:
@clebertsuconic does that answer your question?

Reply:
This simply will be a nightmare to keep this in sync as we add more.